From: Yutao Meng <mengyutao@uniontech.com>
Subject: [PATCH] winex11.drv: Fix crash when ctx is NULL in glxdrv_wglSwapBuffers().
Message-Id: <20220308085704.12679-1-mengyutao@uniontech.com>
Date: Tue,  8 Mar 2022 16:57:04 +0800

The variable ctx can be NULL, and it is checked before every sync_context() function call. However, it is not checked in the ExtEscape() function call in which it is trying to get the hdc member variable of ctx.

Signed-off-by: Yutao Meng <mengyutao@uniontech.com>
---
 dlls/winex11.drv/opengl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winex11.drv/opengl.c b/dlls/winex11.drv/opengl.c
index 11111111111..11111111111 100644
--- a/dlls/winex11.drv/opengl.c
+++ b/dlls/winex11.drv/opengl.c
@@ -3516,7 +3516,7 @@ static BOOL WINAPI glxdrv_wglSwapBuffers( HDC hdc )
     update_window_surface( gl, hwnd );
     release_gl_drawable( gl );
 
-    if (escape.drawable)
+    if (ctx && escape.drawable)
         NtGdiExtEscape( ctx->hdc, NULL, 0, X11DRV_ESCAPE, sizeof(escape), (LPSTR)&escape, 0, NULL );
     return TRUE;
 }

-- 
2.37.2

